pkg/aws/client.go

```diff
@@ -546,27 +546,28 @@ func (c *awsClient) FilterVPCsPrivateSubnets(subnets []*ec2.Subnet) ([]*ec2.Subn
 
 // isPublicSubnet a public subnet is a subnet that's associated with a route table that has a route to an
 // internet gateway
-func (c *awsClient) isPublicSubnet(subnetID *string, routeTables []*ec2.RouteTable) (bool, error) {
+func (c *awsClient) isPublicSubnet(subnetID *string, routeTables []ec2types.RouteTable) (bool, error) {
 	subnetRouteTable, err := c.getSubnetRouteTable(subnetID, routeTables)
 	if err != nil {
 		return false, err
 	}
 
 	for _, route := range subnetRouteTable.Routes {
-		if strings.Contains(aws.StringValue(route.GatewayId), "igw") {
+		if strings.Contains(aws.ToString(route.GatewayId), "igw") {
 			return true, nil
 		}
 	}
 
 	return false, nil
 }
 
-func (c *awsClient) getSubnetRouteTable(subnetID *string, routeTables []*ec2.RouteTable) (*ec2.RouteTable, error) {
+func (c *awsClient) getSubnetRouteTable(subnetID *string,
+	routeTables []ec2types.RouteTable) (*ec2types.RouteTable, error) {
 	// Subnet route table — A route table that's associated with a subnet
 	for _, routeTable := range routeTables {
 		for _, association := range routeTable.Associations {
-			if aws.StringValue(association.SubnetId) == aws.StringValue(subnetID) {
-				return routeTable, nil
+			if aws.ToString(association.SubnetId) == aws.ToString(subnetID) {
+				return &routeTable, nil
```

Comment:
Were you able to test this change? previously we returned `routeTable` and now `&routeTable`